tokio-tls/src/lib.rs

```diff
@@ -20,10 +21,22 @@
 //! built. Configuration of TLS parameters is still primarily done through the
 //! `native-tls` crate.
 
-use futures::{Async, Future, Poll};
-use native_tls::{Error, HandshakeError};
+use native_tls::{Error, HandshakeError, MidHandshakeTlsStream};
+use std::future::Future;
 use std::io::{self, Read, Write};
-use tokio_io::{try_nb, AsyncRead, AsyncWrite};
+use std::marker::Unpin;
+use std::pin::Pin;
+use std::task::{Context, Poll, Waker};
+use tokio_io::{AsyncRead, AsyncWrite};
+
+scoped_tls::scoped_thread_local!(static WAKER: Waker);
+/// A wrapper around types which only implement `AsyncRead` and `AsyncWrite`
+/// to be compatible with native_tls::TlsStream
+#[derive(Debug)]
+pub struct AllowStd<S> {
```

Comment:
Does this need to be `pub`?

Reply:
`TlsStream`'s `get_ref(&self)` returns `&native_tls::TlsStream<AllowStd<S>>` which needs `AllowStd` to be public

Reply:
Let’s remove those fns for now